test/gerkin/tests/features/data_delete.feature

```diff
@@ -0,0 +1,19 @@
+Feature: Remove record from the database.
+  Request endpoint: DELETE /data/{code}/{version}/delete
+
+  Background:
+    Given database with the valid schema has been set up
+
+  Scenario: Successfully delete existing record
+    When I make a DELETE request with a valid payload ID <ID>
+    Then I receive a HTTP 204 response
+
+  Scenario: Failure when delete existing record
+    When I make a DELETE request with a invalid payload
+    Then I receive a HTTP 400 response
```

Comment:
Shouldn't it be 404 Not Found?
Invalid request is not part of API Spec and I think it'd be best to consolidate it. 
Additionally test case specification should be more about behaviour than HTTP Repsonse, e.g. "Information about failed delete operation is received"